### PR TITLE
Run tests under .NET 8

### PR DESCRIPTION
**Pull request by vbreuss**
Discussion in #2465 

Enable tests to run under [.NET 8](https://dotnet.microsoft.com/en-us/download/dotnet/8.0).

## IMPORTANT 

* [ ] If the PR touches the public API, the changes have been approved in a separate issue with the "api-approved" label.
* [x] The code complies with the [Coding Guidelines for C#](https://www.csharpcodingguidelines.com/).
* [x] The changes are covered by unit tests which follow the Arrange-Act-Assert syntax and the naming conventions such as is used [in these tests](../tree/develop/Tests/FluentAssertions.Equivalency.Specs/MemberMatchingSpecs.cs#L51-L430).
* [ ] If the PR adds a feature or fixes a bug, please update [the release notes](../tree/develop/docs/_pages/releases.md) with a functional description that explains what the change means to consumers of this library, which are published on the [website](https://fluentassertions.com/releases).
* [ ] If the PR changes the public API the changes needs to be included by running [AcceptApiChanges.ps1](../tree/develop/AcceptApiChanges.ps1) or [AcceptApiChanges.sh](../tree/develop/AcceptApiChanges.sh).
* [ ] If the PR affects [the documentation](../tree/develop/docs/_pages), please include your changes in this pull request so the documentation will appear on the [website](https://www.fluentassertions.com/introduction).
    * [ ] Please also run `./build.sh --target spellcheck` or `.\build.ps1 --target spellcheck` before pushing and check the good outcome

### CODE REVIEW

**Review by jnyrup — APPROVED** [review]

**Review by dennisdoomen — APPROVED** [review]

**Review by dennisdoomen — CHANGES_REQUESTED** [review]
We should update the documentation (index.html, releases.md, etc) to mention that we're compatible with .NET 8.0

File: docs/index.html [hunk 1]
```diff
@@ -29,7 +29,7 @@
      [<i class="fa fa-coffee"></i> Buy us a coffee](https://ko-fi.com/dennisdoomen){: .btn .btn--paypal}
      <br/>
      A very extensive set of extension methods that allow you to more naturally specify the expected outcome of a TDD or
-     BDD-style unit tests. Targets .NET Framework 4.7, as well as .NET 6, .NET Standard 2.0 and 2.1.'
+     BDD-style unit tests. Targets .NET Framework 4.7, as well as .NET 6, .NET 8, .NET Standard 2.0 and 2.1.'
```
Comment (IT-VBFK) [inline docs/index.html:32]: Should this be just a note that we are ".NET 8 ready", since we don't target .NET 8

Comment (dennisdoomen) [reply docs/index.html:32]: Yes, something like that. Or we should rephrase all of the references to the frameworks. Like "Works with .NET Standard 2.0 and higher, .NET Framework 4.7 and higher and .NET 6 and higher"

Comment (dennisdoomen) [reply docs/index.html]: I also think there are a couple of other places where we mention the version, including the NuGet description and such.

**Comment by dennisdoomen** [comment]
> **103 new problems** were found

This is something of a challenge as well. All the issues reported might be existing ones that now get redetected through .NET 8 target. We may have to update the baseline or bump the limit.

File: docs/index.html (hunk 1 above)
Comment (vbreuss) [reply docs/index.html]: You were right. I found three more places 😄

**Comment by vbreuss** [comment]
> > **103 new problems** were found
> 
> This is something of a challenge as well. All the issues reported might be existing ones that now get redetected through .NET 8 target. We may have to update the baseline or bump the limit.

How would I do this?
- Updating the baseline will have to be done by you, right?
- Bumping the limit would mean, increasing the `failThreshold` from 19 to 122 in [qodana.yaml](https://github.com/fluentassertions/fluentassertions/blob/develop/qodana.yaml#L04)?

**Comment by vbreuss** [comment]
> **103 new problems** were found

I had a look at the mentioned issues in the [Files changed-Tab](https://github.com/fluentassertions/fluentassertions/pull/2466/files) and most of them seem to be incorrect🤷: 
- All issues are only for two files ([AssertionExtensions.cs](https://github.com/fluentassertions/fluentassertions/pull/2466/files#diff-8c66543bec53a3b357f3d29bcf013319d7b7a6eb66586b5bc6380d3f875a203a) and [CallerIdentifier.cs](https://github.com/fluentassertions/fluentassertions/pull/2466/files#diff-efe2c23ca42b295cfce0b45df46d88ee407d966ca0f86b3c2d3f583274ae0ffa))
- The issues list (nearly) all using statements as incorrect.
  image,
- Also the cannot resolve reference in XML comment seem incorrect
  image

**Comment by jnyrup** [comment]
According to https://youtrack.jetbrains.com/issue/QD-7660/Qodana-busted-in-.NET-8-RTM support for .NET 8 is not available yet, but will be with v233.
We're at [QDNET-232.10203.29](https://github.com/fluentassertions/fluentassertions/actions/runs/6919435180/job/18822800870#step:3:60)

**Comment by IT-VBFK** [comment]
Cool.. seems work 😉

**Comment by vbreuss** [comment]
> @vbreuss Can you check what happens if you change `args: --baseline,qodana.sarif.json,--ide,QDNET` to ` args: --baseline,qodana.sarif.json,--ide,QDNET-EAP`? Just to verify this fixes the issues...

Yes, it works.
previously it was
image

now it is
image

*Will now undo the test...*

**Comment by IT-VBFK** [comment]
@vbreuss FYI You could just have dropped the commit 😉

**Review by dennisdoomen — APPROVED** [review]

**Comment by IT-VBFK** [comment]
wow.. the qodana scan after this PR on develop 🙈

**Comment by dennisdoomen** [comment]
@vbreuss can you please put back that -EAP argument?

**Comment by vbreuss** [comment]
> @vbreuss can you please put back that -EAP argument?

Of course, see #2474

**Pull request merged**